src/schema/resolvers.js

```diff
@@ -250,124 +273,245 @@ module.exports = {
+        const tx = await centralizedOracle.bet({
+          contractAddress: oracleAddress,
+          index: optionIdx,
+          amount,
+          senderAddress,
+        });
+        txid = tx.txid;
       } catch (err) {
-        logger.error(`Error insertion db: ${err.message}`);
+        logger.error(`Error calling CentralizedOracle.bet: ${err.message}`);
         throw err;
       }
 
+      // Insert Transaction
+      const tx = {
+        _id: txid,
+        txid,
+        version,
+        type: 'BET',
+        status: 'PENDING',
+        senderAddress,
+        oracleAddress,
+        optionIdx,
+        token: 'QTUM',
+        amount,
+        createdTime: Date.now().toString(),
+      };
+      await DBHelper.insertTransaction(Transactions, tx);
+
       return tx;
     },
 
     setResult: async (root, data, { db: { Transactions } }) => {
-      const version = data.version;
-      const senderAddress = data.senderAddress;
-      const oracleAddress = data.oracleAddress;
-      const resultIdx = data.resultIdx;
-      const consensusThreshold = data.consensusThreshold;
-
-      // rpc call to approve
-      let approveTxid;
+      const {
+        version,
+        topicAddress,
+        oracleAddress,
+        optionIdx,
+        amount,
+        senderAddress,
+      } = data;
+
+      // Make sure allowance is 0, or it needs to be reset
+      let approveAmount;
+      let type;
+      if (await isAllowanceEnough(senderAddress, topicAddress, amount)) {
+        approveAmount = amount;
+        type = 'APPROVESETRESULT';
+      } else {
+        approveAmount = 0;
+        type = 'RESETAPPROVESETRESULT';
+      }
+
+      // Send approve tx
+      let txid;
       try {
-        const resp = await fetch('http://localhost:5555/approve', {
-          method: 'POST',
-          body: JSON.stringify({
-            spender: oracleAddress,
-            value: consensusThreshold,
-            senderAddress,
-          }),
-          headers: { 'Content-Type': 'application/json' },
-        }).then(res => res.json());
-
-        approveTxid = resp.result.txid;
+        const tx = await bodhiToken.approve({
+          spender: topicAddress,
+          value: approveAmount,
+          senderAddress,
+        });
+        txid = tx.txid;
       } catch (err) {
-        logger.error(`Error call /approve: ${err.message}`);
+        logger.error(`Error calling BodhiToken.approve: ${err.message}`);
         throw err;
       }
 
+      // Insert Transaction
       const tx = {
-        _id: approveTxid,
+        _id: txid,
+        txid,
         version,
-        type: 'APPROVESETRESULT',
-        txStatus: 'PENDING',
+        type,
+        status: 'PENDING',
         senderAddress,
-        entityId: oracleAddress,
-        optionIdx: resultIdx,
-        token: 'QTUM',
-        amount: consensusThreshold,
+        topicAddress,
+        oracleAddress,
+        optionIdx,
+        token: 'BOT',
+        amount,
```

Comment:
convert amount from `decimalstring` to `hexstring`

Reply:
same note as above